presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -2125,4 +2127,64 @@ core::PlanFragment VeloxQueryPlanConverter::toVeloxQueryPlan(
         toJsonString(partitioningHandle));
   }
 }
+
+velox::core::PlanFragment VeloxQueryPlanConverter::toBatchVeloxQueryPlan(
+    const protocol::PlanFragment& fragment,
+    const std::shared_ptr<protocol::TableWriteInfo>& tableWriteInfo,
+    const protocol::TaskId& taskId,
+    const std::shared_ptr<operators::ShuffleInterface>& writerShuffle) {
+  auto planFragment = toVeloxQueryPlan(fragment, tableWriteInfo, taskId);
+
+  // If the last node is a PartitionedOutputNode, it means this fragment ends
+  // with a shuffle stage. We convert the PartitionedOutputNode to a chain of
+  // following nodes:
+  // (1) A PartitionAndSerializeNode.
+  // (2) A "gather" LocalPartitionNode that gathers results from multiple
+  //     threads to one thread.
+  // (3) A ShuffleWriteNode.
+  auto partitionedOutputNode =
+      dynamic_pointer_cast<const core::PartitionedOutputNode>(
+          planFragment.planNode);
+  if (!partitionedOutputNode) {
+    VELOX_CHECK_NULL(
+        writerShuffle,
+        "Writer shuffle interface passed in when there is no "
+        "PartitionedOutputNode.");
+    return planFragment;
+  }
+  VELOX_CHECK_NOT_NULL(
+      writerShuffle,
+      "No writer shuffle interface passed in for converting a "
+      "PartitionedOutputNode.");
+  if (partitionedOutputNode->isBroadcast()) {
+    VELOX_UNSUPPORTED(
+        "Broadcast partitioned output node in batch is currently not "
+        "supported.");
+  }
+  auto partitionAndSerializeNode =
+      std::make_shared<operators::PartitionAndSerializeNode>(
+          "shuffle-partition-serialize",
+          partitionedOutputNode->keys(),
+          partitionedOutputNode->numPartitions(),
+          partitionedOutputNode->sources().back(),
+          partitionedOutputNode->partitionFunctionFactory());
+
+  std::vector<core::PlanNodePtr> sources;
+  sources.push_back(partitionAndSerializeNode);
+  auto localPartitionNode = std::make_shared<core::LocalPartitionNode>(
```

Comment:
```
auto localPartitionNode = std::make_shared<core::LocalPartitionNode>({std::move(partitionAndSerializeNode)})
```

Reply:
I mean you don't need to create a temp sources above? It is solely used by LocalPartitionNode ctor?